basiclti/basiclti-impl/src/java/org/sakaiproject/lti/impl/LTIRoleMapperImpl.java

```diff
@@ -106,25 +104,21 @@ public void setSiteService(SiteService siteService) {
                     M_log.debug("No match, falling back to determine role");
                 }
 
-                String maintainRole = site.getMaintainRole();
-                String joinerRole = site.getJoinerRole();
+                ServerConfigurationService cnf = (ServerConfigurationService) ComponentManager
+                		.get(ServerConfigurationService.class);
+				String maintainRole = site.getMaintainRole();
 
-                for (Role r : roles) {
-                    String roleId = r.getId();
-                    if (maintainRole == null && (roleId.equalsIgnoreCase("maintain") || roleId.equalsIgnoreCase("instructor"))) {
-                        maintainRole = roleId;
-                    }
+		if (maintainRole == null) {
+		    maintainRole = cnf.getString("lti.role.mapping.Instructor", null);
+		}
 
-                    if (joinerRole == null && (roleId.equalsIgnoreCase("access") || roleId.equalsIgnoreCase("student"))) {
-                        joinerRole = roleId;
-                    }
-                }
-
-                boolean isInstructor = ltiRole.indexOf("instructor") >= 0;
-                newRole = joinerRole;
-                if (isInstructor && maintainRole != null) {
-                    newRole = maintainRole;
-                }
+		boolean isInstructor = ltiRole.indexOf("instructor") >= 0;
+		newRole = cnf.getString("lti.role.mapping.Student", null);
+		if (isInstructor && maintainRole != null) {
+		   newRole = maintainRole;
+		}
```

Comment:
Would be slightly more efficient to have this as an if/else statement:
if (isInstructor && maintainRole != null)
   newRole = maintainRole
else
  newRole = cnf.getString();